lockfile/src/parsers/go_sum.rs

```diff
@@ -1,16 +1,27 @@
 use nom::branch::alt;
 use nom::bytes::complete::{tag, take_until};
 use nom::character::complete::{alphanumeric1, line_ending, space0, space1};
-use nom::combinator::{opt, recognize};
-use nom::multi::many1;
+use nom::combinator::recognize;
+use nom::multi::{many0, many1};
 use nom::sequence::{preceded, tuple};
 use phylum_types::types::package::PackageType;
 
-use super::IResult;
+use crate::parsers::{take_till_line_end, IResult};
 use crate::{Package, PackageVersion};
 
 pub fn parse(input: &str) -> IResult<&str, Vec<Package>> {
-    let (input, mut pkgs) = many1(package)(input)?;
+    let (input, lines) = many0(preceded(space0, take_till_line_end))(input)?;
+
+    let mut pkgs = Vec::new();
+    for line in lines {
+        if line.contains("/go.mod") {
+            continue;
+        }
```

Comment:
This seems pretty crude... Are we certain that `/go.mod` can't show up anywhere else?

It seems like it would be safer to write something like this:

```rs
let (input, pkgs) = many1(package)(input)?;

// Filter out "/go.mod" hashes
let pkgs = pkgs.into_iter().filter(|p| !p.version.ends_with("/go.mod")).collect();
```

Meanwhile, the `package_version()` parser would be updated like so:

```rs
fn package_version(input: &str) -> IResult<&str, &str> {
    // Take away any leading whitespace.
    let (input, _) = space0(input)?;

    // Accept all of `v[a-zA-Z0-9.+-]+` with an optional "/go.mod" suffix
    let (input, version) = recognize(tuple((
        tag("v"),
        many1(alt((alphanumeric1, tag("."), tag("-"), tag("+")))),
        opt(tag("/go.mod")),
    )))(input)?;

    // Expect at least one whitespace after version.
    let (input, _) = space1(input)?;

    Ok((input, version))
}
```